### ISSUE TITLE
Frame records need to be linked to painting records

### ISSUE THREAD

**Issue by edgartdata** [ycba-cia/blacklight-collections2#82]
Frame records have the accession number of the paintings they are on, appended with FR.

For example, the frame B1977.14.67FR https://collections.britishart.yale.edu/vufind/Record/3414343 goes with the painting B1977.14.67 https://collections.britishart.yale.edu/vufind/Record/1669236

This logic is true for 99% of the frames. What to do for when it is not true?

**Comment by edgartdata** [comment]
@EdwardTown1  
There are other objects that are related in TMS but disassociated in Vufind that it would be helpful to reunite in Blacklight. For example, https://collections.britishart.yale.edu/vufind/Record/3651628 is an album that contains 80 prints, one of which is https://collections.britishart.yale.edu/vufind/Record/3661442

Another example is the Turner sketchbook: https://collections.britishart.yale.edu/vufind/Record/2069877 which is linked in TMS to its individual pages, such as https://collections.britishart.yale.edu/vufind/Record/2069903

Albums and sketchbooks are linked to their pages through the TMS parent/child relationship.

**Comment by EdwardTown1** [comment]
Yes, would be fantastic if we could reunite these prints/watercolours

**Comment by edgartdata** [comment]
@edgartdata to correct frame-painting relatioship in TMS and create a new LIDO section.

**Comment by edgartdata** [comment]
@edgartdata Give Eric a list of painting-frame to add to the SOLR index.

**Comment by yulgit1** [comment]
blacklight for above examples:
Turner album: http://10.5.96.187:3000/catalog/tms:4110
Turner sketchbook: http://10.5.96.187:3000/catalog/tms:57847

Scultori album:http://10.5.96.187:3000/catalog/tms:24562
Scultori print: http://10.5.96.187:3000/catalog/tms:56142

create linkages based on accession numbers?
aggregate image in iiif? 
    issue: mixed cds image levels (jpg,jp2)
    issue: manifest or info.json(osd) level?

**Comment by yulgit1** [comment]
@edgardata @EdwardTown1 
I added "Link to Frame" and "Linked to Framed Image" conversely below "Collection" in the tombstone data

Resolved in:
https://github.com/ycba-cia/blacklight-collections2/commit/b2c8a2bec139f00b5f0e2b177838584b40965565
https://github.com/ycba-cia/blacklight-collections2/commit/7dd092c1dd6d3441ca6f0a42e3ea38a367d8aea0

**Comment by edgartdata** [comment]
@edgartdata to make sure that all links to frames from BlackLight painting records are appropriate. Give Eric a list of frames whose accession numbers are not predicated on the paintings.

**Comment by yulgit1** [comment]
Reopening this issue to add lookup table to map obj2frame and frames2obj.

Table creation:
https://git.yale.edu/ermadmix/camel_collections2/wiki/frames-lookup-table
To be modified as necessary to remap like:
`update frames_mapping set FrameObjectID = 72360, FrameObjectNumber = 'B1977.14.70FR-2' where ObjectNumber = 'B1977.14.70`

Modification of indexing:
https://git.yale.edu/ermadmix/camel_collections2/commit/570d10c07d6760a8ce9e686d9cbe5ed40ae097b5
https://git.yale.edu/ermadmix/camel_collections2/commit/2e697ece808937f0804b6a66aaf50281daadcdaf
https://git.yale.edu/ermadmix/camel_collections2/commit/c47f8a8eb1ea8816a024bbfeec661273724dc33c
https://git.yale.edu/ermadmix/camel_collections2/commit/f26b5da7a73905e1fc47807f88d67f6299c908b3
https://git.yale.edu/ermadmix/camel_collections2/commit/fffc011ac6f51b167030e4a1edb900d844f5acfa

Modifications of blacklight:
https://github.com/ycba-cia/blacklight-collections2/commit/0aa77c7d52d9e2a91adc58d82d36260b7c33f7aa

**Comment by edgartdata** [comment]
Thanks @yulgit1.
I see that you have associated/linked the 3 FR-2 frames with the paintings that they currently are on: 
B1977.14.70FR-2, B1981.25.310FR-2, and B2009.2FR-2. Super!
Going forward, do I need to give you a list of changes, or can I do it myself? I don't have access to the creation table.

**Comment by yulgit1** [comment]
If there are any changes if you give them to me in this form:

`update frames_mapping set FrameObjectID = 72360, FrameObjectNumber = 'B1977.14.70FR-2' where ObjectNumber = 'B1977.14.70'`

I can just run that, it will only take a few seconds.  If I get a chance I may extend that collmgmt, used for deletions so you can do this yourself.  (BTW, was also thinking of something along these lines for the marc locnaf mapping too).

**Comment by edgartdata** [comment]
[like]  Delmas-Glass, Emmanuelle reacted to your message:
________________________________

**Comment by yulgit1** [comment]
@edgartdata please see:

https://collmgmt.britishart.yale.edu/frames/index
